packages/lit-dev-content/site/tutorials/content/carousel/02.md

```diff
@@ -0,0 +1,58 @@
+The carousel is now showing its first child item but there's no way
+to change what's displayed. It needs a `selected` property to control which
+item is shown.
+
+Add the `selected` property accessor and accompanying code above the `render`
+method.
+
+### motion-carousel
+
+{% switchable-sample %}
+
+```ts
+private selectedInternal = 0;
+@property({type: Number})
+selected = 0;
+
+get maxSelected() {
+  return this.childElementCount - 1;
+}
+
+hasValidSelected() {
+  return this.selected >= 0 && this.selected <= this.maxSelected;
+}
+
+render() {
+  if (this.hasValidSelected()) {
+    this.selectedInternal = this.selected;
+  }
+  // ...
+```
+
+```js
+static properties = { selected: {type: Number} };
+
+selectedInternal = 0;
+selected = 0;
+
+get maxSelected() {
+  return this.childElementCount - 1;
+}
+
+hasValidSelected() {
+  return this.selected >= 0 && this.selected <= this.maxSelected;
+}
+
+render() {
+  if (this.hasValidSelected()) {
+    this.selectedInternal = this.selected;
+  }
```

Comment:
Gonna let this one go but we can revisit. Generally computed state should be done in `willUpdate`, yes, but this isn't a hard rule and in this case, the values are directly related to rendering/animation. To use `willUpdate` effectively, it would make sense to compute `left` there as well, but that would require storing more state (the `delta`), which doesn't seem worth it.